### PR TITLE
show more specific error messages to students when they attempt to vi…

### PR DESCRIPTION
**Pull request by tanyaparker**
…ew restricted content

image

### CODE REVIEW

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -588,13 +588,14 @@ def redirect_under_13_without_tos_teacher(level)
     return false unless level.game == Game.applab || level.game == Game.gamelab
 
     if current_user && current_user.under_13? && current_user.terms_version.nil?
-      redirect_to '/', :flash => { :alert => I18n.t("errors.messages.too_young") }
+      error_message = current_user.student_of_any_teacher? ? I18n.t("errors.messages.teacher_must_accept_terms") : I18n.t("errors.messages.too_young")
```
Comment (ashercodeorg) [inline dashboard/app/helpers/levels_helper.rb]: Honestly I think `student.teachers.any?` is far more clear than `student_of_any_teacher?` (also less symbols). My vote is to eliminate the helper method, though feel welcome to disagree.

File: dashboard/config/locales/en.yml [hunk 2]
```diff
@@ -23,7 +23,9 @@ en:
   errors:
     messages:
       not_saved: 'Please correct the following errors:'
-      too_young: 'Your teacher must accept the updated Terms of Service before you can access this content. Your teacher can do this by logging into Code Studio and clicking the "Review Updated Terms" button near the top of Code Studio homepage.'
+      too_young: 'This content has age restrictions in place and is not available for younger students.'
+      teacher_must_accept_terms: 'Your teacher must accept the updated Terms of Service before you can access this content. Your teacher can do this by logging into Code Studio and clicking the "Review updated terms" button near the top of the Code Studio homepage.'
```
Comment (ashercodeorg) [inline dashboard/config/locales/en.yml]: Does it make sense to explain to the student the workflow for the teacher?

**Comment by ashercodeorg** [comment]
LGTM (see comments).

File: dashboard/config/locales/en.yml (hunk 2 above)
Comment (tanyaparker) [reply dashboard/config/locales/en.yml]: Probably @poorvasingal 's thinking was that the student would just show their screen to the teacher? But I kind of agree with you. Would suggest something like "This content has age restrictions in place and is not available for younger students. Ask your teacher to accept the latest Terms of Service to give you access to this content."

What if a teacher wants to say no, is that awkward?

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (tanyaparker) [reply dashboard/app/helpers/levels_helper.rb]: Oh for sure, I'll fix.

File: dashboard/config/locales/en.yml (hunk 2 above)
Comment (tanyaparker) [reply dashboard/config/locales/en.yml]: Discussed with @poorvasingal offline. Yes, the intention was for students to probably show teachers the error message. Teachers may ignore the interstitial when it pops up and not realize that it gives students access to content, so this error message gives them more information on how it's related. I'll rephrase it a bit, but it at least gives a little more helpful information to the teacher. We'll also watch the data for what % of teachers don't accept the latest terms and we can do more to try and increase it like making the banner more prominent or showing the interstitial on more pages.

**Pull request merged**